### DISCUSSION TITLE
Welcome to meross_lan Discussions!

### DISCUSSION THREAD

**Discussion by krahabb** [krahabb/meross_lan#3] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

About myself:
I'm an hardware/software engineer with passion for automation and DIY little projects. I grew up developing c++ software in a software company and later moved to independent mobile development for Android (java) and iOS (objective-c) backed by .NET (C#) cloud services
Still, I'm in love with c++ (even though I don't code it anymore since 10 years) but I find myself messing up with Python and the likes for the sake of making homeassistant an enjoyable tool.
I have installed it in my home a few months ago and it is working great. Nevertheless I like to play with it and put my hands in the code
Also, I find this cloud dependant IoT devices a pain for a lot of reasons and I'm working on integrating all of them (or at least the most) in my local environment.

Thank you for contributing or just reading :)

**Comment by Topolynx** [comment]
HI @krahabb, thanks for your great work!
I'm Antonio, sound engineer with very little competence in code ;)
I'm tired too of my devices depending by their cloud. expecially my Meross ones that tends to become unavailable quite 40-50 times per day...
I absolutely need your help. I've installed your meross_lan integration and managed to get my MSS310 plugs work the HTTP way.
I have noticed that in the HTTP way the plugs are a bit laggy (20-25 seconds) updating their status, so i want to switch to MQTT.
I have mosquitto already installed in docker and I've used it with no tls. Before going on, I would explain what I'm doing with MQTT: it is a "bridge" between my Homeassistant Core and Node-Red; I have an odd Tuya climate that I control with Node-Red, so Node-Red publishes it as an MQTT device, which in turn is integrated in Home Assistant.
Today I wanted switch to MQTT TLS for re-pair my Meross plugs, so I've followed the [@bytespider's wiki](https://github.com/bytespider/Meross/wiki/MQTT). While the protocol change was simple in Node-Red, it is a pain in Home Assistant.
I've tried almost every configuration possible but I can't configure the MQTT integration correctly, 
Homeassistant's log reports:
`Failed to connect to MQTT server due to exception: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1125)
`
while mosquitto log reports:
`1629978907: New connection from 192.168.1.7:44503 on port 8883.
1629978907: OpenSSL Error[0]: error:14037418:SSL routines:ACCEPT_SR_KEY_EXCH:tlsv1 alert unknown ca
1629978907: OpenSSL Error[1]: error:140370E5:SSL routines:ACCEPT_SR_KEY_EXCH:ssl handshake failure
1629978907: Client <unknown> disconnected: Protocol error.`
What should I do?

**Reply by krahabb** [reply]
Hello @Topolynx ,
welcome aboard !)
You too falling in MQTT hell 😱
Btw, I cheated it a bit in my own setup so I can't really tell how to solve the TLS issue. 
But, my setup works and it is like this: 
- mosquitto broker vanilla installation on the same raspi as HA core over venv (no HASSIO at all)
- mosquitto setup with 2 listeners:
- -1 for non TLS traffic which I use to connect HA
- -2 for TLS with self signed CA and certificate using the IP as common name
- mosquitto users: 1 for HA
- the 2nd listener has no user/pass configured and allows anonymous
- per-listener-setting is a must so you can trim down mqtt configuration depending on your client needs: just add a new listener with own cfg so to allow different auth-tls-whatever personalization without the need to do 1 cfg do it all
- remember even meross plugs allow you to set the mqtt port so you're not really restricted to use the 'standard' 1883 for non TLS and 8883 for TLS

**Reply by krahabb** [reply]
with this setup I don't have to configure any meross device user in mqtt since it's 'password file' is not present and allowing anonymous will bypass the user authentication

I don't remember if I had to set acl too since when I set it up they were likely not very strict by default and I haven't upgraded my setup since then

I remember upgrading the 1st time was a pain with mosquitto not being able to even start so I reverted and banned it from sudo apt upgrade ;)

**Reply by Topolynx** [reply]
Thank you @krahabb, I didn't know I could configure multiple ports with different settings each 😸.
As soon as I can I'll try to re-pair my plugs, hoping not to disturb you with help requests :)

**Comment by boba51** [comment]
Hi,
I'm Bob, I've been involved in IT in various roles over the years, but never programming, currently on the Networking side and mainly Cisco in my current role.

Fairly new to Home Assistant although have tried it a few times in the past without great success, currently installed on a VM running on Synology NAS and of course have number of Meross Smartplugs hence my use of your integration.

I'll start a discussion topic on my issue in the hope someone can point me in the right direction.

Cheers,

Bob

**Comment by gazunni** [comment]
Hello

I just started getting an error on some of my Meross Light bulbs "Failed to perform the action light/turn_on. 'capacity'.   I have reloaded the integration, rebooted home assistant and turned the light on and off from the native Meross application, which generally has solved any issues in the past. Any thoughts?

**Reply by krahabb** [reply]
Hello @gazunni,
Could you provide model and version of the Meross bulb exposing the problem ? It might be new devices have changed a bit since this kind of issues never appeared in the past.

Better would be to have a complete [diagnostic](https://github.com/krahabb/meross_lan/wiki/Diagnostics#download-diagnostics) for the device.

**Reply by krahabb** [reply]
I've also opened a specific [issue](https://github.com/krahabb/meross_lan/issues/553) for this to track

**Reply by gazunni** [reply]
Model number msl100r
Firmware version 2.1.4
Hardware version 2.0.0

**Reply by krahabb** [reply]
Well...it is one of the first generations so it should work...anyway I think I know where the issue could lie but I'd still need a diagnostic for this specific device in order to try understand why and where it could be so different (I've used msl120 mainly to develop the light interface) .

**Reply by gazunni** [reply]
Hello

Thanks for replying.  I have tried multiple times to download the diagnostic file but I receive an error, "failed to download file error: The network connection was lost"
I can download a diagnostic file on any other working device without issue.  It appears that since HA cannot communicate with the device it can't download the diagnostic file.

**Reply by krahabb** [reply]
If you're using any proxy (even nabucasa cloud forwarding) that might happen because the process of creating the diagnostic is a bit too long to be managed by the browser connection (or something like that: tbh I don't know really why it doesnt reliably work )

I have made some adjustments anyway to the code so that the next release could possibly address the issue without further hassle.
Just let me a couple of days to finish sorting things up for the release ;)
